src/routes/CheckIn/modules/check-in.js

```diff
@@ -1,27 +1,47 @@
 // ------------------------------------
 // Constants
 // ------------------------------------
+export const UPDATE_PRIMARY_INFO = 'UPDATE_PRIMARY_INFO'
+export const UPDATE_PRIMARY_INFO_SUCCESS = 'UPDATE_PRIMARY_INFO_SUCCESS'
+export const UPDATE_PRIMARY_INFO_FAIL = 'UPDATE_PRIMARY_INFO_FAIL'
 
 // ------------------------------------
 // Actions
 // ------------------------------------
-export const actions = {
+export function updatePrimaryInfo (fields) {
+  return {
+    type: UPDATE_PRIMARY_INFO,
+    payload: fields,
+  }
+}
 
+
+export const actions = {
+  updatePrimaryInfo,
 }
 
 // ------------------------------------
 // Action Handlers
 // ------------------------------------
 const ACTION_HANDLERS = {
-
+  [UPDATE_PRIMARY_INFO]: (state, action) => {
+    const fields = action.payload
+    const newState = {}
+
+    for (let field in fields) {
+      if (fields[field].touched && fields[field].value) {
+        newState[field] = fields[field].value
+      }
+    }
+
+    return Object.assign({}, state, newState)
```

Comment:
It works but it's preferable to keep state immutable, so we should create new objects instead.